### PR TITLE
docs: rewrite built-in directives section of Template Syntax

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

1. Rewritten documentation on the Built-in Directives section prose.
2. Rebuilt example that is standalone so that it can be its own example when Built-in Directives is pulled out as its own doc.
3. Example is more accessible (added/edited markup to help screen readers).
4. Example is now item themed rather than hero themed.
5. New e2e tests for the example.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

This is the 7th PR of rewrites of the Template Syntax page. They all work now in Template Syntax and and are prepped for separating Template Syntax into smaller docs, each with its own example.

### CODE REVIEW

File: aio/content/guide/template-syntax.md [hunk 1]
```diff
@@ -1115,116 +1115,99 @@ Fortunately, the Angular [_NgModel_](guide/template-syntax#ngModel) directive is
...
-
-<code-example path="template-syntax/src/app/app.component.html" region="event-binding-1" header="src/app/app.component.html" linenums="false">
-</code-example>
-
-You still benefit from directives that simplify complex tasks.
-Angular still ships with built-in directives; just not as many.
-You'll write your own directives, just not as many.
-
-This segment reviews some of the most frequently used built-in directives,
+Angular offers two kinds of built-in directives; attribute
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: You are correct. Changed!

File: aio/content/guide/template-syntax.md [hunk 2]
```diff
@@ -1115,116 +1115,99 @@ Fortunately, the Angular [_NgModel_](guide/template-syntax#ngModel) directive is
...
 and the [`FormsModule`](guide/forms "Forms") define their own attribute directives.
-This section is an introduction to the most commonly used attribute directives:
-
-* [`NgClass`](guide/template-syntax#ngClass) - add and remove a set of CSS classes
-* [`NgStyle`](guide/template-syntax#ngStyle) - add and remove a set of HTML styles
-* [`NgModel`](guide/template-syntax#ngModel) - two-way data binding to an HTML form element
+The most common attribute directives are as follows:
 
+* [`NgClass`](guide/template-syntax#ngClass)&mdash;adds and remove a set of CSS classes.
+* [`NgStyle`](guide/template-syntax#ngStyle)&mdash;adds and remove a set of HTML styles.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: Good catch - fixed.

File: aio/content/guide/template-syntax.md [hunk 3]
```diff
@@ -1115,116 +1115,99 @@ Fortunately, the Angular [_NgModel_](guide/template-syntax#ngModel) directive is
...
+both initially and when the dependent properties change.
 
 </div>
 
 <hr/>
 
 {@a ngStyle}
 
-### NgStyle
+### `NgStyle`
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: None at all. Fixed! :)

File: aio/content/guide/template-syntax.md [hunk 4]
```diff
@@ -1328,25 +1297,29 @@ Here are all variations in action, including the uppercase version:
 ## Built-in _structural_ directives
 
 Structural directives are responsible for HTML layout.
-They shape or reshape the DOM's _structure_, typically by adding, removing, and manipulating
+They shape or reshape the DOM's structure, typically by adding, removing, and manipulating
 the host elements to which they are attached.
 
+This section is an introduction to the common built-in structural directives:
+
+* [`NgIf`](guide/template-syntax#ngIf)&mdash;conditionally add or remove an element from the DOM.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: Hm, food for thought. Even so, `<ng-container>` and `<ng-template>` would both be elements in the DOM and most elements could have any number of child nodes, right? (Really, correct me here if I'm wrong). Either way, would you have a suggestion for better wording? I definitely want clarity and precision and trust your judgement. :)

[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: Wow @trotyl , thank you for your thoughtful explanation. I've tried incorporating your points, but am not sure I have it just right, especially on the part about moving `Angular translates...`.

Would you want to submit a PR to my fork? You don't have to be concerned about the nuances of language, I can handle that part. I'd be happy to collaborate.

**Comment by jenniferfell** [comment]
@kapunahelewong Hi. See https://github.com/angular/angular/issues/27564. Can you work in something here about ngif else? Or make sure there are links to the API ref pages (which Judy has updates for in a separate PR)? Thanks!

**Comment by kapunahelewong** [comment]
@jenniferfell You bet.

File: aio/content/guide/template-syntax.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: Resolving this one for now, and will revisit. Hoping to get this landed and then do a more granular pass in the second phase of the Template Syntax restructure.

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Caretaker note: The Codefresh job is unrelated and still pending. I am using a global approval

**Comment by AndrewKushnir** [comment]
@kapunahelewong, it looks like this PR now has a conflict with master branch after merging PR 25561 that also updated .github/CODEOWNERS file. Could you please rebase this PR and resolve conflict(s)? Feel free to add "merge" label back once PR is updated. Thank you.

**Pull request closed without merging**